src/tools/crawler/main.rs

```diff
@@ -232,16 +232,14 @@ async fn main() {
                 if crawler.should_connect(addr) {
                     let crawler_clone = crawler.clone();
                     tokio::spawn(async move {
-                        if crawler_clone.connect(addr).await.is_ok() {
-                            sleep(Duration::from_secs(1)).await;
-                            let _ = crawler_clone.unicast(addr, Message::GetAddr);
-                        }
+                        // Don't care about the result - errors are logged inside connect() and
+                        // successful connections do not involve any further actions as sending
+                        // GetAddr message was moved to process_message in protocol.rs.
+                        let _ = crawler_clone.connect(addr).await;
```

Comment:
We don't need to document historical reasons, we can just point out where the new solution is and how it works shortly
```suggestion
                        // Once the Version message is received in the process_message function,
                        // GetAddr will be requested from the peer
                        let _ = crawler_clone.connect(addr).await;
```

Reply:
Changed.